apps/builder/app/builder/features/style-panel/sections/borders/border-style.tsx

```diff
@@ -0,0 +1,28 @@
+import type { StyleProperty } from "@webstudio-is/css-data";
+import type { RenderCategoryProps } from "../../style-sections";
+import { BorderProperty } from "./border-property";
+
+const borderPropertyOptions = {
+  borderTopStyle: {},
+  borderRightStyle: {},
+  borderLeftStyle: {},
+  borderBottomStyle: {},
+} as const satisfies Partial<{ [property in StyleProperty]: unknown }>;
```

Comment:
need to remember this thing

Reply:
isn't value known though?

Reply:
Here I just check that property names are from StyleProperty values.
The value of property will be checked by react component. 
But no way to check on Component side that no additional properties will be added.

Example:

https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAKjgQwM5wEoFNkGN4BmUEIcA5FDvmQNwBQduEAdqvAMImTNbPwC8cABRhiYVAC44AbwhgYwFpMxYmUACYAeMsjJwAPuQBGZADRxmAVxBGsUAHxwAvgEo4-R9LpwfcSjEsoZgtLABtQ+idGJXg5BSV3GWQJAEZzIyk0uFxM8ywpACYnFHQmVhh6OgB6KrgAOQg4O2IoOAALOyw6TU5wFl5Y+UVWRNkhpWKq+wYauAAJTrgASTgAcyx4ZGDm6GjyuDjh1ALR5KyMuCycy7zC4rRsmLhUZAVUAmAsdGw1LR09QxkEzmSzMADWzAgAHdmPZ6EA

So with satisfies I want to show that I don't mind what values this component has, moreover don't want to create types for values, but keys must be real StyleProperty.

Also satisfies forces intellisense.